### PR TITLE
add navigation menu + refactor

### PR DESCRIPTION
**Pull request by IlyesBenAmara**
This PR adds a navigation menu to the navigation bar + a refactor to the shell component
image

### CODE REVIEW

File: src/components/shell/NavigationMenu.tsx [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+            >
+              John Schmidt
+            </Typography>
+            <Typography
+              fontWeight={400}
+              fontSize={'14px'}
+              lineHeight="143%"
+              letterSpacing={'0.17px'}
+            >
+              johnshmidt@match.com
```
Comment (Lumiere-Lodi) [inline src/components/shell/NavigationMenu.tsx:56]: I know we are not there yet, How about passing a variable already to cater for when the API sends the user info? Or even use the mock backend for it? What do you think?

**Review by Lumiere-Lodi — COMMENTED** [review]
LGTM. I left a question in there. If That is not needed for now then the PR is good to go.

File: src/components/shell/NavigationMenu.tsx (hunk 1 above)
Comment (marrouchi) [reply src/components/shell/NavigationMenu.tsx:56]: @IlyesBenAmara You can mock a API endpoint "GET /users/me" that would return a user object : {id, username, email}

Comment (marrouchi) [reply src/components/shell/NavigationMenu.tsx:56]: BWT there's a useAuth() hook that is supposed to return the user object. So you need to get the current user with the AuthContextProvider

Comment (IlyesBenAmara) [reply src/components/shell/NavigationMenu.tsx:56]: Understood, working on it :technologist:

File: src/components/shell/NavigationBar.tsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
+import {
+  Dashboard as DashboardIcon,
+  People as PeopleIcon,
+  Search as SearchIcon
+} from '@mui/icons-material'
+import { AppBar, Box, Button, Toolbar, Typography } from '@mui/material'
+import NavigationMenu from './NavigationMenu'
+
+const barColour =
```
Comment (marrouchi) [inline src/components/shell/NavigationBar.tsx:9]:
```suggestion
const colorGradient =
```

File: src/components/shell/NavigationBar.tsx [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        >
+          <Typography variant="h6" sx={{ color: '#FFC400' }}>
+            Je
+          </Typography>
+          <Typography
+            variant="h6"
+            sx={{ color: theme => theme.palette.text.primary }}
+          >
+            MPI
+          </Typography>
```
Comment (marrouchi) [inline src/components/shell/NavigationBar.tsx:30]: Let's refactor this into a "Logo" component ;)

File: src/components/shell/NavigationMenu.tsx [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+import * as React from 'react'
+import { useAuth } from '../../hooks/useAuth'
+
+interface NavigationMenuProp {}
+
+const NavigationMenu: React.FC<NavigationMenuProp> = () => {
+  const { user } = useAuth()
+
+  const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null)
+  const open = Boolean(anchorEl)
```
Comment (marrouchi) [inline src/components/shell/NavigationMenu.tsx:14]:
```suggestion
  const isOpen = Boolean(anchorEl)
```

File: src/components/shell/NavigationMenu.tsx [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+import MenuItem from '@mui/material/MenuItem'
+import * as React from 'react'
+import { useAuth } from '../../hooks/useAuth'
+
+interface NavigationMenuProp {}
+
+const NavigationMenu: React.FC<NavigationMenuProp> = () => {
+  const { user } = useAuth()
+
+  const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null)
```
Comment (marrouchi) [inline src/components/shell/NavigationMenu.tsx:13]: Shouldn't this be a useRef() instead ?

Comment (marrouchi) [reply src/components/shell/NavigationMenu.tsx:13]: I'm not sure why we store the html element in a state ? Could we achieve this another way ?

**Review by marrouchi — CHANGES_REQUESTED** [review]
Awesome, just a couple of code excellency comments.

File: src/components/shell/NavigationMenu.tsx (hunk 5 above)
Comment (IlyesBenAmara) [reply src/components/shell/NavigationMenu.tsx:13]: We are following MUI documentation on using the Menu component 
https://mui.com/material-ui/react-menu/

**Review by marrouchi — APPROVED** [review]

**Review by Lumiere-Lodi — APPROVED** [review]
LGTM

**Pull request merged**